docs/spec/spec.md

```diff
@@ -50,40 +50,214 @@ to interface with a relational database such as MySQL, MSSQL, Postgresql and Ora
+       # This can be changed through the configuration API with the `ballerina.sql.transactionIsolation` key
+       TransactionIsolation? transactionIsolation = transactionIsolation;
+       # SQL query to validate a connection. Leave unset to use driver-native validation. The default value is
+       # nil. This can be changed through the configuration API with the `ballerina.sql.connectionTestQuery` key
+       string? connectionTestQuery = connectionTestQuery;
+       # SQL executed when a new connection is created. Useful for session-level settings. The default value is
+       # nil. This can be changed through the configuration API with the `ballerina.sql.connectionInitSql` key
+       string|string[]? connectionInitSql = connectionInitSql;
+       # Marks connections as read-only by default. Use with care when apps issue writes. The default value is
+       # false. This can be changed through the configuration API with the `ballerina.sql.readOnly` key
+       boolean readOnly = readOnly;
+       # Allows suspending the pool for maintenance. Rarely needed; keep disabled. The default value is false.
+       # This can be changed through the configuration API with the `ballerina.sql.allowPoolSuspension` key
+       boolean allowPoolSuspension = allowPoolSuspension;
+       # Isolates pool's internal queries from application transactions. The default value is false. This can
+       # be changed through the configuration API with the `ballerina.sql.isolateInternalQueries` key
+       boolean isolateInternalQueries = isolateInternalQueries;
    |};
+
+   # Represents the transaction isolation level.
+   public enum TransactionIsolation {
+       # Read committed isolation level
+       TRANSACTION_READ_COMMITTED,
+       # Read uncommitted isolation level
+       TRANSACTION_READ_UNCOMMITTED,
+       # Repeatable read isolation level
+       TRANSACTION_REPEATABLE_READ,
+       # Serializable isolation level
+       TRANSACTION_SERIALIZABLE
+   }
    ```
 
+### Connection Pool Configuration Examples
+
+With the enhanced configuration options, you can now fine-tune the connection pool for different scenarios:
+
+**High-traffic application with connection leak detection:**
+```ballerina
+sql:ConnectionPool connPool = {
+    maxOpenConnections: 50,
+    connectionTimeout: 10.0,
+    leakDetectionThreshold: 60.0,  // Log potential leaks after 1 minute
+    keepAliveTime: 30.0           // Keep connections alive every 30 seconds
+};
+```
+
+**Read-only reporting application:**
+```ballerina
+sql:ConnectionPool connPool = {
+    maxOpenConnections: 10,
+    readOnly: true,
+    transactionIsolation: TRANSACTION_READ_COMMITTED,
+    connectionTestQuery: "SELECT 1"
+};
+```
+
+**Development environment with custom initialization:**
+```ballerina
+sql:ConnectionPool connPool = {
+    maxOpenConnections: 5,
+    poolName: "DevPool",
+    connectionInitSql: ["SET session sql_mode = 'STRICT_TRANS_TABLES'", "SET time_zone = '+00:00'"]
```

Comment:
is it possible to ask init SQL query as Ballerina parameterized query type?

Reply:
I'll check.

Reply:
Yes, +1 to go with the current approach. This init script only used for validating the connections.